Pagina de alteração de senha e Mudanças de Layout

TTOExtreme opened the pull request:
-- Adicionado pagina de alteração de senha
-- Adicionado Menu do Usuário
-- Adicionado icones para as paginas

Commit:
Adicionado pagina user

Adicionado pagina 'user' contendo o formulário de mudança de senha, com confirmação de senha.

Commit:
Adicionado pagina user

Adicionado pagina 'user' contendo o formulário de mudança de senha, com confirmação de senha.

Commit:
Fix Layout Dropdown e icons

--Adicionado Dropdown na tela ao passar com o mouse em cima da "engrenagem"
-- Movido o botão de alteração de senha para o menu de dropdown
-- Adicionado icones em svg para os botões
-- movido tela user para changepass pois só contem a parte de alteração de senha

Commit:
Fix Layout Dropdown e icons

--Adicionado Dropdown na tela ao passar com o mouse em cima da "engrenagem"
-- Movido o botão de alteração de senha para o menu de dropdown
-- Adicionado icones em svg para os botões
-- movido tela user para changepass pois só contem a parte de alteração de senha

GCastilho reviewed (changes requested):
Coloca texto em volta dos ícones

Pelo menos no ícone de mudar senha (dentro da lista)

E deixa a lista visível, ela tá com a mesma cor do fundo, não dá pra ver

GCastilho commented:
Tem erros do eslint tbm, roda um eslint-fix, pq se não o código não compila

Commit:
Alterado visual Dropdown

-- Alterado cor do fundo
-- Adicionado Titulo antes do icone

Commit:
Alterado visual Dropdown

-- Alterado cor do fundo
-- Adicionado Titulo antes do icone

GCastilho committed:
Refatorado o Svelte

-- Refatorado o código do frontend, retirando algumas inconsistências sem alterar funcionalidades

GCastilho committed:
Refatorado o Svelte

-- Refatorado o código do frontend, retirando algumas inconsistências sem alterar funcionalidades

GCastilho committed:
Merge branch 'master' into User-page-utils

GCastilho committed:
Refatorado handler do request de alteração de senha

-- Refatorado o handler do request de alteração de senha
-- Refatorado mensagens de erro de alteração de senha do forntend

GCastilho committed:
Refatorado handler do request de alteração de senha

-- Refatorado o handler do request de alteração de senha
-- Refatorado mensagens de erro de alteração de senha do forntend

GCastilho commented on src/router/index.js:
```diff
@@ -10,4 +10,7 @@ Router.use('/register', require('./register'))
 
+/**@description Handler de todos os requests para /user */
+Router.use('/changepass', require('./user'))
+
```
Coloca esse request de changepass sub um /user. Ele ficando ele "/user/changepass"

GCastilho reviewed (changes requested):
Uma modificação simples dos endereços

joaojvds committed:
Implementação da HTTP REST API (#51)

Refactor do webserver:
-- A pasta router agora é a pasta base do web server inteiro, não apenas do roteador
-- Adicionado um handler para subdomínios, requests para um subdomínio inexistente retornam 404
-- Requests sem um subdomínio são roteados à pasta root

HTTP API
-- Adicionado subdomínio 'api' para requests à httpApi
-- Adicionado endpoint /v1 à API
-- Implementado uma api para requests HTTP no padrão REST para substituir o websocket em requests que não precisem ser em tempo real

Geral:
-- O método getBalance da User agora pode retornar os saldos em string ou Decimal128
-- Libs que simplesmente expandiam types do typescript foram movidas para uma pasta @types, a pasta libs será apenas para módulos que expandem funções de um módulo
-- Adicionado type para permitir referência à user no request do express
-- Resolvido um bug no script de compilação que impedia a compilação em sistemas linux
-- Implementados testes para a HTTP API

Co-authored-by: GCastilho <castilho.gabriel14@gmail.com>

Commit:
Mudança de rotas e layout

-- Modificado rota para acesso a mudança de senha
-- Alterado layout que ficava escondido em telas reduzidas
-- Reduzido tamanho da caixa de "Dropdown"

Commit:
Merge remote-tracking branch 'origin/User-page-utils' into User-page-utils

GCastilho committed:
Cleanup

-- Removido user extra
-- Minor changes

GCastilho committed:
Adicionado testes para atualização de password

-- Adicionado testes de atualização de password
-- Removido arquivos de lixo deixados pelo TTO
-- Minor changes and fixes

GCastilho committed:
Minor update

-- Renomeado testede requests de login para se adequar ao fato de que ele não mais testa apenas o login

GCastilho reviewed (approved):
Agora o professor não pode mais falar q não tem essa função básica
